src/utils/FileUpload.ts

```diff
@@ -0,0 +1,124 @@
+import fs from "fs";
+import path from "path";
+import { IncomingForm, Files, File } from "formidable";
+import { Request } from "express";
+import Formidable from "formidable/Formidable";
+import { R4 as fhir } from "@ahryman40k/ts-fhir-types";
+import FhirResourceBuilder from "@/utils/FhirResourceBuilder";
+import { v4 as uuid } from "uuid";
+
+export default class FileUpload {
+  private static uploadDirectory(req: Request): string {
+    const context = req.params.context;
+    const fhirId = req.params.fhirId;
+    return process.env.NODE_ENV === "development"
+      ? `./${context}/${fhirId}`
+      : `/${context}/${fhirId}`;
+  }
+
+  private static createDirIfNotExists(path: string) {
+    if (!fs.existsSync(path)) {
+      fs.mkdirSync(path, { recursive: true });
+    }
+  }
+
+  private static changeFilePath(file: File, uploadDirectory: string): void {
+    if (!FileUpload.filetypeIsWhitelisted(file)) return;
+    const fileExtension = file.path.split(".").pop();
+    const newPath = `${uploadDirectory}/${uuid()}.${fileExtension}`;
+    // this prevents directory traversal attacks
+    // where "../" is used to traverse other directories
+    if (newPath.startsWith(uploadDirectory)) {
+      file.path = newPath;
+    }
+  }
+
+  private static initializeUploadForm(uploadDirectory: string): Formidable {
+    const form = new IncomingForm({
+      keepExtensions: true,
+      multiples: false,
+      allowEmptyFiles: false,
+      minFileSize: 1,
+      maxFieldsSize: 0, // = unlimited
+      maxFileSize: 4000 * 1024 * 1024, // = 4 gb
+    });
+    form.on("fileBegin", (name, file) => this.changeFilePath(file, uploadDirectory));
+    return form;
+  }
+
+  private static async parseRequest(req: Request, form: Formidable): Promise<Files> {
+    return new Promise(function (resolve, reject) {
+      form.parse(req, function (err, fields, files) {
+        if (err !== null) reject(err);
+        else resolve(files);
+      });
+    });
+  }
+
+  private static filetypeIsWhitelisted(file: File): boolean {
+    return file.path.match(/\.(mp4|jpg|json|pdf|ply|png)$/i) ? true : false;
+  }
+
+  private static urlOf(path: string): string {
+    const protocol =
+      process.env.NODE_ENV === "development" || process.env.INSECURE ? "http://" : "https://";
```

Comment:
Lets reduce that to:
```suggestion
    const protocol =
      process.env.NODE_ENV === "development" ? "http://" : "https://";
```